File: backend/src/main/java/com/travel/entity/Article.java
package com.travel.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@TableName("articles")
public class Article {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long userId;
    
    private String title;
    
    private String coverImage;
    
    private String content;
    
    private String spotIds;
    
    private String tags;
    
    private Integer viewCount;
    
    private Integer likeCount;
    
    private Integer collectCount;
    
    private Integer commentCount;
    
    private Integer status;  // 0待审核 1已发布 2已下架
    
    private Integer isTop;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    @TableLogic
    private Integer deleted;
}

File: frontend/src/api/article.ts
import request from './axios'

export interface Article {
  id: number
  userId: number
  title: string
  coverImage: string
  content: string
  tags: string
  viewCount: number
  likeCount: number
  collectCount: number
  commentCount: number
  status: number
  isTop: number
  createdAt: string
}

// 游记列表
export function getArticleListApi(params: { page: number; pageSize: number; status?: number }) {
  return request.get('/article/list', { params })
}

// 热门游记
export function getHotArticlesApi(limit = 10) {
  return request.get('/article/hot', { params: { limit } })
}

// 游记详情
export function getArticleDetailApi(id: number) {
  return request.get(`/article/${id}`)
}

// 发布游记
export function createArticleApi(data: Partial<Article>) {
  return request.post('/article', data)
}

// 我的游记
export function getMyArticlesApi(params: { page: number; pageSize: number }) {
  return request.get('/article/my', { params })
}

// 点赞游记
export function likeArticleApi(id: number) {
  return request.post(`/article/${id}/like`)
}
